libsolidity/ast/Types.cpp

```diff
@@ -3403,12 +3416,27 @@ MemberList::MemberMap MagicType::nativeMembers(ContractDefinition const*) const
 				StateMutability::Pure
 			)}
 		});
-	default:
-		solAssert(false, "Unknown kind of magic.");
+	case Kind::TypeMeta:
+	{
+		solAssert(
+			m_typeArgument && m_typeArgument->category() == Type::Category::Contract,
+			"Only contracts supported for now"
+		);
+		ContractDefinition const& contract = dynamic_cast<ContractType const&>(*m_typeArgument).contractDefinition();
+		if (contract.annotation().unimplementedFunctions.empty() && contract.constructorIsPublic())
```

Comment:
Hm. Perhaps it makes sense making this a function on `ContractDefinition` so that it has proper documentation?